internal/ethapi/api_test.go

```diff
@@ -88,6 +90,24 @@ func TestTransaction_RoundTripRpcJSON(t *testing.T) {
 	}
 }
 
+func TestTransaction_RoundTripRpcJSON(t *testing.T) {
+	var (
+		config = params.AllEthashProtocolChanges
+		tests  = allTransactionTypes(common.Address{0xde, 0xad}, config)
+	)
+	testTransactionMarshal(t, tests, config)
+}
+
+func TestTransactionBlobTx(t *testing.T) {
+	newUint64 := func(val uint64) *uint64 { return &val }
+	config := *params.TestChainConfig
+	config.ShanghaiTime = newUint64(0)
+	config.CancunTime = newUint64(0)
```

Comment:
```suggestion
	config := *params.TestChainConfig
	config.ShanghaiTime = new(uint64)
	config.CancunTime = new(uint64)
```
Isn't this sufficient?

Reply:
thanks